### PR TITLE
Catch invalid coordinates in HGVS names

### LINKED ISSUE #17: Catch invalid HGVS names like NC_000005.10:g.177421339_177421327delACTCGAGTGCTCC
NC_000005.10:g.177421339_177421327delACTCGAGTGCTCC appears in ClinVar, and is an invalid name (the genomic start/stop coords are not in increasing order). This causes `parse_hgvs_name` to raise an IndexError. It should raise InvalidHGVSName instead

### PR DESCRIPTION
**Pull request by ChrisBeaumont**
Closes #17 

I'd also like to bump the version so we can use it in website, but Counsyl's PyPI lists the most recent version as 0.9.3 whereas `setup.py` is currently set to 0.9.2. Is there another version of this repo that we should be syncing up with?

### CODE REVIEW

File: pyhgvs/__init__.py [hunk 1]
```diff
@@ -970,6 +970,13 @@ def parse_genome(self, details):
 
         raise InvalidHGVSName(details, 'genomic allele')
 
+    def _validate(self):
+        """
+        Check for internal inconsistencies in representation
+        """
+        if self.start > self.end:
```
Comment (pkaleta) [inline pyhgvs/__init__.py]: I didn't find any place in the [hgvs spec](http://www.hgvs.org/mutnomen/recs-DNA.html#del) that disallows having the same start and stop, like `78_78delC`, so I think that this is fine as is. `78delC` might be a preferred way to encode such a change.

File: setup.py [hunk 2]
```diff
@@ -32,7 +33,7 @@ def main():
         scripts=[],
         install_requires=['pip>=1.2'],
         tests_require=[str(line.req) for line in
-                       parse_requirements('requirements-dev.txt')],
+                       parse_requirements('requirements-dev.txt', session=PipSession())],
```
Comment (pkaleta) [inline setup.py]: Is this something that's required by the newer pip?

**Comment by pkaleta** [comment]
This LGTM.

File: setup.py (hunk 2 above)
Comment (ChrisBeaumont) [inline setup.py]: From what I gather, yes (I get an error otherwise).

**Comment by ChrisBeaumont** [comment]
@pkaleta do you know anything about the version discrepancy noted above?

**Comment by mdrasmus** [comment]
> @pkaleta do you know anything about the version discrepancy noted above?

I pushed some more commit to the public repo. You should be good to remerge master now.

**Comment by ChrisBeaumont** [comment]
Ok this is good to merge. @pkaleta can you hit the button, since I don't have write access?

**Comment by pkaleta** [comment]
Thanks @ChrisBeaumont!

**Pull request merged**